timeinterval/interval.go

```diff
@@ -0,0 +1,134 @@
+package timeinterval
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// Interval describes an interval which can be bounded by a startsAt and/or an endsAt time.
+// If startsAt is unset it will be interpreted as "unbounded" (goes infinitely long back in time).
+// If endsAt is unset it will be interpreted as "unbounded" (goes infinitely long into the future).
+// If both startsAt and endsAt is unset, then it will span all of time and be fairly pointless ;-)
+type Interval struct {
+	startsAt *time.Time
+	endsAt   *time.Time
+	duration *time.Duration
```

Comment:
This is here because we need to support intervals specified by duration like:

1) StartsAt = Time & EndsAt = Time
2) StartsAt = Time & EndsAt = duration after StartsAt
3) StartsAt = duration before EndsAt & EndsAt = Time

If we want to marshal/unmarshal to ISO8601 without loosing the original format it was specified in, then we need to have three fields here: startsAt, endsAt and duration.